File: RepoTransBench/target_projects/Python/Go/rust-lang_homu/tests/original/test_comments_test.go
package original

import (
	"strings"
	"testing"
)

// Simulated stripMention removes the first @mention (case-insensitive "homu") and leading/trailing spaces, returns rest
func stripMention(s string) string {
	trimmed := strings.TrimSpace(s)
	lower := strings.ToLower(trimmed)
	if strings.HasPrefix(lower, "@homu") {
		return strings.TrimSpace(trimmed[len("@homu"):])
	}
	tokens := strings.Fields(trimmed)
	if len(tokens) > 0 && strings.HasPrefix(strings.ToLower(tokens[0]), "@homu") {
		return strings.TrimSpace(strings.Join(tokens[1:], " "))
	}
	if strings.HasPrefix(trimmed, "@") {
		// Remove first word
		toks := strings.Fields(trimmed)
		if len(toks) > 1 {
			return strings.Join(toks[1:], " ")
		}
		return ""
	}
	return trimmed
}

func TestStripMention(t *testing.T) {
	msg := "@homu r+"
	expected := "r+"
	if got := stripMention(msg); got != expected {
		t.Errorf("stripMention(%q) = %q, want %q", msg, got, expected)
	}
	msg = " @Homu  approve please! "
	expected = "approve please!"
	if got := stripMention(msg); got != expected {
		t.Errorf("stripMention(%q) = %q, want %q", msg, got, expected)
	}
}

File: RepoTransBench/target_projects/Python/C++/rust-lang_homu/tests/original/test_comments_test.cpp
#include <gtest/gtest.h>
#include <string>
#include <regex>
// #include "comments.h"

// Strip @homu or @username prefix (with optional spaces or casing) and leading/trailing spaces
std::string strip_mention(const std::string& msg) {
    std::regex mention_re("^\\s*@\\w+\\s+");
    return std::regex_replace(msg, mention_re, "");
}

TEST(CommentsTest, StripMention) {
    std::string msg1 = "@homu r+";
    EXPECT_EQ(strip_mention(msg1), "r+");
    std::string msg2 = " @Homu  approve please! ";
    EXPECT_EQ(strip_mention(msg2), "approve please!");
}